lib/core/job_cleanup.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'core'
+require 'utility/logger'
+
+module Core
+  class JobCleanUp
+    class << self
+      def execute(connector_id = nil)
+        process_orphaned_jobs
+        process_stuck_jobs(connector_id)
+      end
+
+      private
+
+      def process_orphaned_jobs
+        Utility::Logger.info('Start cleaning up orphaned jobs...')
+        orphaned_jobs = ConnectorJob.orphaned_jobs
+        if orphaned_jobs.empty?
+          Utility::Logger.info('No orphaned jobs found. Skipping...')
+          return
+        end
+
+        # delete content indicies in case they are re-created by sync job
+        content_indices = orphaned_jobs.map(&:index_name).compact.uniq
+        ElasticConnectorActions.delete_indices(content_indices) if content_indices.any?
+        result = ConnectorJob.delete_jobs(orphaned_jobs)
+        Utility::Logger.error("Error found when deleting jobs: #{result['failures']}") if result['failures']&.any?
+        Utility::Logger.info("Successfully deleted #{result['deleted']} out of #{result['total']} orphaned jobs.")
```

Comment:
I think it'll make it better if we log here anything only if something was really deleted, to not produce too many logs.